src/gui/editorwidgets/qgslistwidgetfactory.cpp

```diff
@@ -44,5 +44,5 @@ QgsEditorConfigWidget *QgsListWidgetFactory::configWidget( QgsVectorLayer *vl, i
 unsigned int QgsListWidgetFactory::fieldScore( const QgsVectorLayer *vl, int fieldIdx ) const
 {
   const QgsField field = vl->fields().field( fieldIdx );
-  return ( field.type() == QVariant::List || field.type() == QVariant::StringList ) && field.subType() != QVariant::Invalid ? 20 : 0;
+  return ( field.type() == QVariant::List || field.type() == QVariant::StringList || field.typeName().compare( QLatin1String( "json" ) ) == 0 || field.typeName().compare( QLatin1String( "jsonb" ) ) == 0 ) && field.subType() != QVariant::Invalid ? 20 : 0;
```

Comment:
Just wondering, should this be case insenstive?

Reply:
I should never be uppercase. But I can add it to be sure here.

Reply:
I took the `==` that is sensitive, because it should be always lowercase, and so it's the same comparison like it's done on other types and there is no confusion with `compare`. In case it should be case intensive, we should change all the comparisons. I could do that, but I don't see importance here...